enhancements/olm/single-ownnamespace-enhancement-proposal.md

```diff
@@ -0,0 +1,418 @@
+
+The enhancement introduces a configuration mechanism through the ClusterExtension API that allows users to specify a `watchNamespace` parameter, enabling operators to be installed in either SingleNamespace mode (watching a namespace different from the install namespace) or OwnNamespace mode (watching the same namespace where the operator is installed).
+
+## Motivation
+
+OLM v0 provided four install modes (AllNamespaces, MultiNamespace, SingleNamespace, OwnNamespace) as part of its multi-tenancy features. While OLM v1 deliberately simplified this by only supporting AllNamespaces mode to avoid the complexity and problems associated with multi-tenancy, there exists significant operator content in catalogs that only supports Single and OwnNamespace modes.
+
+### User Stories
+
+#### Story 1: Legacy Operator Migration
+As a cluster administrator migrating from OLM v0 to OLM v1, I want to install operators that only support SingleNamespace or OwnNamespace install modes so that I can continue using existing operator content without requiring operator authors to modify their bundles.
+
+#### Story 2: Security-Conscious Deployments
+As a security-conscious administrator, I want to install operators that explicitly limit their scope to specific namespaces so that I can maintain strict RBAC boundaries and reduce the attack surface of operator deployments.
+
+#### Story 2: Operator Author Requirements
+As an operator developer, I have operators designed for namespace isolation that only support Single or OwnNamespace modes, and I want my customers to be able to deploy these operators in OpenShift with OLM v1 so that I can maintain the intended isolation boundaries while benefiting from OLM v1's simplified management.
+
+### Goals
+
+- Enable installation of registry+v1 bundles that only support SingleNamespace or OwnNamespace install modes
+- Provide a configuration mechanism through the ClusterExtension API to specify watch namespaces
+- Maintain backward compatibility with existing OLM v0 operator content
+- Generate appropriate RBAC resources scoped to the target namespaces
+- Support seamless migration path for customers moving from OLM v0 to OLM v1
+
+### Non-Goals
+
+- Re-introducing multi-tenancy features or supporting multiple installations of the same operator
+- Supporting MultiNamespace install mode (watching multiple namespaces)
+- Modifying the fundamental OLM v1 architecture or adding complex multi-tenancy logic
+- Supporting install mode switching after initial installation as a first-class feature
+- Enabling this feature for bundle formats other than registry+v1
+
+## Proposal
+
+Update the OLMv1 operator-controller to:
+1. Validate bundle compatibility with the requested install mode during resolution
+2. Parse the `watchNamespace` configuration from ClusterExtension.spec.config.inline
+3. Determine the install mode based on the relationship between install namespace and watch namespace
```

Comment:
4. Validate the watchNamespace configuration against the install modes supported by the bundle